airtime_mvc/application/forms/LiveStreamingPreferences.php

```diff
@@ -125,12 +138,12 @@ public function updateVariables()
                 array ('ViewScript',
                     array (
                       'viewScript'                  => 'form/preferences_livestream.phtml',
-                      'master_source_host'          => isset($masterSourceParams["host"])?$masterSourceParams["host"]:"",
-                      'master_source_port'          => isset($masterSourceParams["port"])?$masterSourceParams["port"]:"",
-                      'master_source_mount'         => isset($masterSourceParams["path"])?$masterSourceParams["path"]:"",
-                      'show_source_host'            => isset($showSourceParams["host"])?$showSourceParams["host"]:"",
-                      'show_source_port'            => isset($showSourceParams["port"])?$showSourceParams["port"]:"",
-                      'show_source_mount'           => isset($showSourceParams["path"])?$showSourceParams["path"]:"",
+                      'master_source_host'          => isset($masterSourceHost)?$masterSourceParams["host"]:"",
+                      'master_source_port'          => isset($masterSourcePort)?$masterSourceParams["port"]:"",
+                      'master_source_mount'         => isset($masterSourceMount)?$masterSourceParams["path"]:"",
+                      'show_source_host'            => isset($showSourceHost)?$showSourceParams["host"]:"",
+                      'show_source_port'            => isset($showSourcePort)?$showSourceParams["port"]:"",
+                      'show_source_mount'           => isset($showSourceMount)?$showSourceParams["path"]:"",
                       'isDemo'                      => $isDemo,
                     )
```

Comment:
Python seems to have a [`urlparse`](https://docs.python.org/2/library/urlparse.html) module as well. I need to step through the file generation if I want to make any sense of it as I don't currently really understand what (and why) it is doing. It looks rather convoluted and I have a hard time believing that it can't be done in a simpler fashion.